api/services/SiteCreator.js

```diff
@@ -73,7 +73,7 @@ const createSiteFromExistingRepo = ({ siteParams, user }) => {
   const { owner, repository } = siteParams
 
   return Site.findOne({
-    where: { owner, repository },
+    where: { owner: owner.toLowerCase(), repository: repository.toLowerCase() },
```

Comment:
That seems reasonable! Anything that lets us see "we lowercase the owner/repo coming in when creating and looking for sites.